onboaring.md

```diff
@@ -0,0 +1,345 @@
+
+Publish your package as you normally would:
+
+```sh-session
+npm version <newversion> 
+
+npm publish
+```
+
+This will trigger a build in Warehouse and publish the module to the `write` endpoint you specified. Initially the build of this version
+will only be available in `DEV`. Subsequently the files are only uploaded to the `DEV` CDN as well. A hash is generated
+from the file content that the browser will use to do caching based on file content. Warehouse provides
+various mechanics to retrieve the build files. We'll [dive into the mechanisms
+later](#using-build-output).
+
+
+#### Monitoring build progress
+
+To check the status of the build process you can either use the [Warehouse UI][ui] or
+[CLI](cli).
+Once you have `Warehouse UI` set up, it will show the
+build progress for each environment. Wait for the `DEV` build to complete.
+
+#### Promote your module to TEST/PROD
+
+After the initial publish, the build will only be available in the `DEV` environment. Assuming the bundle
+works as expected, it can be promoted to `TEST` and/or `PROD`. Promoting your module's bundle to a specific
+environment can be done through `npm dist-tag`. For example, to promote your bundle to `TEST` run:
+
+```
+npm dist-tag add [module-name]@[semver] test --reg=https://wherever-you-deployed-warehouse.ai
+```
+
+Where `[module-name]` is the package name of your module and `[semver]` the exact semver that was published to
+`DEV`. Also be aware that the `--reg` argument is required, since `publishConfig` only applies to `npm publish`.
+To promote your module to `PROD`, simply `npm dist-tag` again but replace `test` with `prod`.
+
+### 5. Using build output
+
+Now that you can promote your module's bundle to each environment, you can start using the bundle.
+First, let's inspect the available bundle output. Then, we'll reference the assets from your web application.
+
+#### Inspect your bundle
+
+[TODO] The easiest way to quickly inspect a build and files on CDN is by using the Warehouse CLI. Install the CLI by
+following [the directions from the readme][cli].
+
+Now let's inspect your build by running the following command. Replace `[module-name]` with your package name.
+This will return bundle and build details for your package in `dev`. Similarly you can inspect your bundle in
+`test` and `prod`.
+
+```sh-session
+wrhs get:build [module-name] dev
+```
+
+#### Serving assets from a web service
+
+Use the [Warehouse-client][client] to retrieve bundle information for your webserver's responses. The client
+requires the same authentication and endpoint details as provided to the [CLI](#inspect-your-bundle).
+Currently, we only have a client available for Node.JS. Usage is straightforward. Install the module as
+a `dependency` of your webserver, e.g. `npm i --save warehouse.ai-api-client`.
+
+Next, setup an instance of the client like below. Ideally the `[username]` and `[password]` are stored in a secure config, for example by using
```

Comment:
we should probably reword this and fixup the example below a bit since the auth is optional based on configuration of the `warehouse.ai` instance.